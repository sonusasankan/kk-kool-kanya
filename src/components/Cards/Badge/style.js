import styled from 'styled-components';

export const BadgeWrapper = styled.div`
    ${props => ((props.top || props.right || props.bottom || props.left) ? 'position: absolute' : '')};
    ${props => (props.top ? 'top: 1rem' : '')};
    ${props => (props.right ? 'right: 1rem' : '')};
    ${props => (props.bottom ? 'bottom: 1rem' : '')};
    ${props => (props.left ? 'left: 1rem' : '')};
    z-index: 2;
`;

export const Badge = styled.div`
    display: inline-block;
    height: 24px;
    width: 24px;  
    position: relative;
    img{
        width: 100%;
    }
    margin-left: -10px;
    &:nth-child(1){
        margin-left: 0;
    }
`;
