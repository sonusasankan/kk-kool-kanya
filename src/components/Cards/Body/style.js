import styled from 'styled-components';

const Body = styled.div`
  padding: 1.5rem;
  background-color: #fff;
  width: 100%;
  &.overlay{
    position: absolute;
    top: 0;
    left: 0;
    height: 100%;
    width: 100%;
    background-color: rgba(0,0,0,0);
    padding: 0;
    display: flex;
    align-items: flex-end;
    .kk-card__text-wrapper{
      padding:  16px !important;
      background-color: rgba(0,0,0,0)!important;
      color: #fff;
      p{
        margin-bottom: 0;
      }
    }
    .kk-card__title{
      margin: 0;
      font-size: 1.125rem;
    }
  }
`;

export default Body;
