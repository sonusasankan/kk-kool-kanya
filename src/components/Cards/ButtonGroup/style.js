import styled from 'styled-components';

const ButtonGroup = styled.div`
    display: flex;
    margin: 0 -16px;
    position: relative;
    bottom: -8px;
    border-radius: 0 0 4px 4px;
    overflow: hidden;
    button{
        flex-grow: 1;
        display: block;
        margin: 0;
        border-radius: 0;
        font-size: 12px;
        padding: 8px 0;
        text-align: center;
        min-height: 34px;
        ${props => (props.split ? `max-width: ${100 / props.split}%` : '')};
    }
`;

export default ButtonGroup;
