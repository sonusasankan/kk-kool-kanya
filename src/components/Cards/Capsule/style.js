import styled from 'styled-components';
import Color from '../../../themes/color';

const Capsule = styled.div`
    height: 72px;
    border-radius: 36px;
    padding: 16px;
    display: inline-flex;
    align-items: center;
    font-height: 1.125rem;
    background-color: #efeff5;
    cursor: pointer;
    .capsule{
        &__title{
            margin-left: 1rem;
        }
        &__img__wrapper{
            height: 44px;
            width: 44px;
            border-radius: 44px;
            overflow: hidden;
        }
    }
    &.selected{
        background-color: ${Color.brandColor};
        color: #FFF;
    }
`;

export default Capsule;