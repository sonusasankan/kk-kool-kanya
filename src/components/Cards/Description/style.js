import styled from 'styled-components';

import Color from '../../../themes/color';

const Description = styled.p`
    margin-top: 16px;
    margin-bottom: 0;
    line-height: 1.44;
    color: #4f565e;
    font-size: 1rem;
    .readMore{
        display: inline;
        color: ${Color.brandColor};
        font-weight: 600;
        cursor: pointer;
    }    
`;

export default Description;
