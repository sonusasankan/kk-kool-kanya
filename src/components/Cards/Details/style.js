import styled from 'styled-components';

import Color from '../../../themes/color';


export const DetailsWrapper = styled.div`
    display: flex;
    margin-bottom: 8px;
    align-items: center;
    // justify-content: ${props => (props.center? "center": "left" )}
`;

export const Details = styled.div`
    color: ${Color.fontLight};
    font-size: 0.875rem;
    text-align: ${props => (props.center? "center": "left" )}
`;

export const Icon = styled.span`
    color: ${Color.fontLight};
    margin-right: 4px;
`;
