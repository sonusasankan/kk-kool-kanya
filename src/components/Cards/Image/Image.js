import React from 'react';
import PropTypes from 'prop-types';

import Image from './style';


const image = ({
  src,
  alt,
  children
}) => (
  <Image className="card__img">
    <img src={src} alt={alt} />
    {children}
  </Image>
);

image.propTypes = {
  alt: PropTypes.string,
  src: PropTypes.string,
  children: PropTypes.func
};

image.defaultProps = {
  alt: '',
  children: []
};

export default image;
