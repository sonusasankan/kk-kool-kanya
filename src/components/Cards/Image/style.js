import styled from 'styled-components';

const Image = styled.div`
    position: relative;
    z-index: 0;
    img{
        width: 100%;
    }
    &::after{
        content: '';
        position: absolute;
        top: 0;
        left: 0;
        height: 100%;
        width: 100%;
        border-radius: 4px 4px 0 0;
        background-image: linear-gradient(rgba(0, 0, 0, 0.75) 0%,rgba(255, 255, 255, 0) 40%);
        transition: all 0.2s ease-out;
        z-index: 1;
    }
`;

export default Image;
