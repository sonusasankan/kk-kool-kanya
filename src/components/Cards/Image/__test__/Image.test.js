import React from 'react';
import { shallow } from 'enzyme';

import { default as CardImage } from '../Image';

import '../../../../assets/icons/style.css';
import styled from 'styled-components';

const Wrapper = styled.div`
    max-width: 260px;
`;

describe('Component', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(
      <Wrapper>
        <CardImage src="https://loremflickr.com/260/220" alt="Alternate text for image" />
      </Wrapper>,
    );
    expect(component).toMatchSnapshot();
  });
});
