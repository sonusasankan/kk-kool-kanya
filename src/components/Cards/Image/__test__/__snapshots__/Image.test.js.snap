// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cmoponent should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef>
    <image
      alt="Alternate text for image"
      src="https://loremflickr.com/260/220"
    />
  </ForwardRef>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <image
        alt="Alternate text for image"
        src="https://loremflickr.com/260/220"
      />,
      "forwardedClass": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": true,
          "rules": Array [
            "
    max-width: 260px;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "forwardedRef": null,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "alt": "Alternate text for image",
        "src": "https://loremflickr.com/260/220",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <image
          alt="Alternate text for image"
          src="https://loremflickr.com/260/220"
        />,
        "forwardedClass": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
    max-width: 260px;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "forwardedRef": null,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "alt": "Alternate text for image",
          "src": "https://loremflickr.com/260/220",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Component should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef>
    <image
      alt="Alternate text for image"
      src="https://loremflickr.com/260/220"
    />
  </ForwardRef>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <image
        alt="Alternate text for image"
        src="https://loremflickr.com/260/220"
      />,
      "forwardedClass": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": true,
          "rules": Array [
            "
    max-width: 260px;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "forwardedRef": null,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "alt": "Alternate text for image",
        "src": "https://loremflickr.com/260/220",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <image
          alt="Alternate text for image"
          src="https://loremflickr.com/260/220"
        />,
        "forwardedClass": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
    max-width: 260px;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "forwardedRef": null,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "alt": "Alternate text for image",
          "src": "https://loremflickr.com/260/220",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
